core/src/main/java/org/elasticsearch/gateway/PrimaryShardAllocator.java

```diff
@@ -289,8 +285,8 @@ private NodesToAllocate buildNodesToAllocate(ShardRouting shard, RoutingAllocati
      * are added to the list. Otherwise, any node that has a shard is added to the list, but entries with highest
      * version are always at the front of the list.
      */
-    NodesAndVersions buildNodesAndVersions(ShardRouting shard, boolean matchAnyShard, Set<String> ignoreNodes,
-                                           AsyncShardFetch.FetchResult<TransportNodesListGatewayStartedShards.NodeGatewayStartedShards> shardState) {
+    NodesResult buildVersionBasedNodes(ShardRouting shard, boolean matchAnyShard, Set<String> ignoreNodes,
```

Comment:
Question - do we have a test that takes indices from 2.x and makes sure we can import them? there should be, but you can you double check?